src/components/SpectralProfiler/SpectralProfilerSettingsPanelComponent/SpectralProfilerSettingsPanelComponent.tsx

```diff
@@ -79,6 +82,70 @@ export class SpectralProfilerSettingsPanelComponent extends React.Component<Widg
         this.widgetStore.setUseWcsValues(changeEvent.target.checked);
     };
 
+    handleXMinChange = (ev) => {
+        if (ev.type === "keydown" && ev.keyCode !== KEYCODE_ENTER) {
+            return;
+        }
+
+        const val = parseFloat(ev.currentTarget.value);
+        const widgetStore = this.widgetStore; 
+        const minX = parseUndefinedValue(widgetStore.minX, widgetStore.linePlotInitXYBoundaries.minXVal);
+        const maxX = parseUndefinedValue(widgetStore.maxX, widgetStore.linePlotInitXYBoundaries.maxXVal);
+        if (isFinite(val) && val !== minX && val < maxX) {
+            widgetStore.setXBounds(val, maxX);
+        } else {
+            ev.currentTarget.value = minX;
+        }
+    };
+
+    handleXMaxChange = (ev) => {
```

Comment:
there's a lot of repetition between these functions. I wonder if it's possible to improve that by extracting a function that takes an event and some parameters? Perhaps that would be needlessly complicated, so treat this as purely a suggestion to investigate